garak/generators/ollama.py

```diff
@@ -3,16 +3,20 @@
 from typing import List, Union
 
 import backoff
-import ollama
 
 from garak import _config
 from garak.attempt import Message, Conversation
+from garak.exception import GeneratorBackoffTrigger
 from garak.generators.base import Generator
 from httpx import TimeoutException
 
 
 def _give_up(error):
-    return isinstance(error, ollama.ResponseError) and error.status_code == 404
+    return (
+        str(error.__class__) == "<class 'ollama._types.ResponseError'>"
+        and hasattr(error, "status_code")
+        and error.status_code == 404
+    )
```

Comment:
How brittle is this this check?  I suspect it may be better to check the negative case since this class only allows one other backoff type:


```suggestion
    return (
        not isinstance(error.__cause__, TimeoutException)
        and hasattr(error.__cause__, "status_code")
        and error.__cause__.status_code == 404
    )
```